File: ios/SsdidWallet/Domain/Verifier/Offline/CredentialRepository.swift
import Foundation

/// Persistent storage for held (received) verifiable credentials.
protocol CredentialRepository {
    func saveCredential(_ credential: VerifiableCredential) async throws
    func getHeldCredentials() async -> [VerifiableCredential]
    func getUniqueIssuerDids() async -> [String]
    func deleteCredential(credentialId: String) async throws
}

File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/domain/verifier/offline/CredentialRepository.kt
package my.ssdid.sdk.domain.verifier.offline

import my.ssdid.sdk.domain.model.VerifiableCredential

interface CredentialRepository {
    suspend fun saveCredential(credential: VerifiableCredential)
    suspend fun getHeldCredentials(): List<VerifiableCredential>
    suspend fun getUniqueIssuerDids(): List<String>
    suspend fun deleteCredential(credentialId: String)
}
